sklearn/model_selection/_search.py

```diff
@@ -513,84 +518,139 @@ def _fit(self, X, y, labels, parameter_iterable):
+                                  self.fit_params, return_parameters=False,
                                   error_score=self.error_score)
-          for parameters in parameter_iterable
+          for parameters in candidate_params
           for train, test in cv.split(X, y, labels))
 
-        # Out is a list of triplet: score, estimator, n_test_samples
-        n_fits = len(out)
-
-        scores = list()
-        grid_scores = list()
-        for grid_start in range(0, n_fits, n_splits):
-            n_test_samples = 0
-            score = 0
-            all_scores = []
-            for this_score, this_n_test_samples, _, parameters in \
-                    out[grid_start:grid_start + n_splits]:
-                all_scores.append(this_score)
-                if self.iid:
-                    this_score *= this_n_test_samples
-                    n_test_samples += this_n_test_samples
-                score += this_score
-            if self.iid:
-                score /= float(n_test_samples)
-            else:
-                score /= float(n_splits)
-            scores.append((score, parameters))
-            # TODO: shall we also store the test_fold_sizes?
-            grid_scores.append(_CVScoreTuple(
-                parameters,
-                score,
-                np.array(all_scores)))
-        # Store the computed scores
-        self.grid_scores_ = grid_scores
-
-        # Find the best parameters by comparing on the mean validation score:
-        # note that `sorted` is deterministic in the way it breaks ties
-        best = sorted(grid_scores, key=lambda x: x.mean_validation_score,
-                      reverse=True)[0]
-        self.best_params_ = best.parameters
-        self.best_score_ = best.mean_validation_score
+        n_candidates = len(candidate_params)
+
+        test_scores, test_sample_counts, _ = zip(*out)
+
+        test_scores = np.array(
+            test_scores, dtype=np.float64).reshape(n_candidates, n_splits)
+        # NOTE test_sample counts (weights) remain the same for all candidates
+        test_sample_counts = np.array(test_sample_counts[:n_splits],
+                                      dtype=np.int)
+
+        scorer_name = _get_scorer_name(self.scorer_)
+        results = dict()
+
+        for split_i in range(n_splits):
+            results["%s_test_split_%d" % (scorer_name, split_i)] = (
+                test_scores[:, split_i])
+
+        # Computed the (weighted) mean and std for all the candidates
+        weights = test_sample_counts if self.iid else None
+        means = np.average(test_scores, axis=1, weights=weights)
+
+        stds = np.sqrt(np.average(
+            (test_scores - means[:, np.newaxis]) ** 2, axis=1,
+            weights=weights))
+
+        results["%s_test_mean" % scorer_name] = means
+        results["%s_test_std" % scorer_name] = stds
+
+        ranks = rankdata(means, method='min')
+        # The ranks now need to be reversed (lower rank for higher scores)
```

Comment:
And if you're going to do this subtraction, `method='max'` is more appropriate, I think